src/util/dbid.cpp

```diff
@@ -0,0 +1,16 @@
+#include "util/dbid.h"
+
+
+const QVariant::Type DbId::kVariantType = DbId().toVariant().type();
+
+//static
+DbId::value_type DbId::valueOf(QVariant /*pass-by-value*/ variant) {
+    if (variant.canConvert<value_type>() &&
+            variant.convert(kVariantType)) {
+        value_type value = variant.value<value_type>();
```

Comment:
OK, bad idea. QString canConvert = true, but may be an empty string with is converted to 0.